File: ccif/src/java/other/org/himalay/ccif/diagnosticReq_1_Mmode.java
// Copyright (2013) Krishna C Tripathi. All rights reserved.
//
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
package org.himalay.ccif ;

import java.io.DataInputStream;
import java.io.DataOutputStream;
import java.io.IOException;
import java.lang.Exception;
import java.io.ByteArrayInputStream;
import java.io.PrintStream;
import org.himalay.msgs.runtime.Created;

import org.himalay.msgs.runtime.*;
@Created(date = "Fri Jan 15 01:33:07 EST 2016")

public  class diagnosticReq_1_Mmode extends   APDUBaseFactory.APDUBase { //Concrete type is diagnosticReq_1_Mmode




    // members variables
    // header
    public APDUHeader header ;
    // length
    public short length ;
    // selfDiagnostics
    public ArrayList<diagnosticReq1mMosemsg> selfDiagnostics ;

    public diagnosticReq_1_Mmode () // throws Exception
    {
        init();
    }

    private void init()
    {
        // Initialize header
        header = new APDUHeader();
        // Initialize length

        // Initialize selfDiagnostics
        selfDiagnostics= new ArrayList<diagnosticReq1mMosemsg>();
        selfDiagnostics.setMemberSize(0);
    }

    public int readNoHeader(DataInputStream istream) throws IOException
    {

        preRead();
        int retVal= 0;
        // read length
        {
            length=(short)(istream.readUnsignedByte());
            retVal+=1;
        }
        // read selfDiagnostics
        for (; istream.available() > 0 ; ) {
            diagnosticReq1mMosemsg temp;
            temp = new diagnosticReq1mMosemsg();
            retVal += temp.read(istream);
            selfDiagnostics.add(temp);
        }

        postRead();
        return retVal;
    }

    public int read(DataInputStream istream) throws IOException
    {
        preRead();
        int retVal= 0;

        // read header
        retVal += header.read(istream);
        // read length
        {
            length=(short)(istream.readUnsignedByte());
            retVal+=1;
        }
        // read selfDiagnostics
        for (; istream.available() > 0 ; ) {
            diagnosticReq1mMosemsg temp;
            temp = new diagnosticReq1mMosemsg();
            retVal += temp.read(istream);
            selfDiagnostics.add(temp);
        }

        postRead();
        return retVal;
    }


    public int write(DataOutputStream ostream) throws IOException
    {
        preWrite();
        int retVal= 0;

        {   /** fix dependent sizes for header **/
        }

        {   /** fix dependent sizes for selfDiagnostics**/
        }

        // write header
        if (header!=null)retVal +=header.write(ostream);
        // write length
        ostream.writeByte(length);
        retVal +=1;
        // write selfDiagnostics
        {
            ArrayList<diagnosticReq1mMosemsg> temp1 = selfDiagnostics;
            for (int iIdx=0; iIdx < temp1.getCount() ; iIdx++) {
                diagnosticReq1mMosemsg temp2    =    temp1.get(iIdx);
                if (temp2!=null)retVal +=temp2.write(ostream);
            }
        }
        postWrite();
        return retVal;
    }

    public int dump(DumpContext dc) throws IOException
    {
        dc.indent();
        dc.getPs().print("diagnosticReq_1_Mmode\n");
        dc.increaseIndent();
        int retVal= 0;
        // write header
        if ( header != null ) {
            dc.indent();
            dc.getPs().println("header") ;
            retVal +=header.dump(dc);
        }
        // write length
        dc.indent();
        dc.getPs().println("length="+length+"(0x"+ Integer.toHexString(length) + ")" );
        // write selfDiagnostics
        {
            ArrayList<diagnosticReq1mMosemsg> temp1 = selfDiagnostics;
            for (int iIdx=0; iIdx < temp1.getCount() ; iIdx++) {
                diagnosticReq1mMosemsg element    = temp1.get(iIdx);
                dc.indent();
                dc.getPs().println(iIdx);
                if ( element != null ) {
                    dc.indent();
                    dc.getPs().println("element") ;
                    retVal +=element.dump(dc);
                }
            }
        }
        dc.decreaseIndent();
        return retVal;
    }


    // Getter for header
    //public APDUHeader getHeader()
    //{
    //    return header ;
    //}


    // Setter for header
    //public void setHeader(APDUHeader val)
    //{
    //    this.header= val;
    //}
    // Getter for length
    //public short getLength()
    //{
    //    return length ;
    //}


    // Setter for length
    //public void setLength(short val)
    //{
    //    this.length= val;
    //}
    // Getter for selfDiagnostics
    //public ArrayList<diagnosticReq1mMosemsg> getSelfDiagnostics()
    //{
    //    return selfDiagnostics ;
    //}


    // Setter for selfDiagnostics
    //public void setSelfDiagnostics(ArrayList<diagnosticReq1mMosemsg> val)
    //{
    //    this.selfDiagnostics= val;
    //}


    public int addToSelfDiagnostics(diagnosticReq1mMosemsg val)
    {
        selfDiagnostics.add(val);
        return selfDiagnostics.size();
    }


    public int removeFromSelfDiagnostics(diagnosticReq1mMosemsg val)
    {
        selfDiagnostics.remove(val);
        return selfDiagnostics.size();
    }


    public int removeNthFromSelfDiagnostics(int idx)
    {
        selfDiagnostics.remove(idx);
        return selfDiagnostics.size();
    }


    public int emptySelfDiagnostics(int idx)
    {
        selfDiagnostics.clear();
        return selfDiagnostics.size();
    }


    public int getSize() throws IOException
    {
        DataOutputStream dos= new DataOutputStream(new NullStream());
        return this.write(dos);
    }


    public void setHeader(APDUHeader header)
    {
        this.header= header;
    }

    public APDUHeader getHeader()
    {
        return this.header;
    }

}

// End of code

File: libccif/src/bp/diagnosticReq_1_Mmode.cpp
//**//
//@Created(date = "Mon Feb 01 22:30:17 EST 2016")
// Copyright (2012) Krishna C Tripathi. All rights reserved.
//
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
//**//
#include "stdio.h"
#include "diagnosticReq_1_Mmode.h"

#ifdef org_himalay_ccif_diagnosticReq_1_Mmode__USE_SMART_PTR
#define ArrayList(x) SmartPtrList<x>
#define ArrayList_iterator(x) SmartPtrList<x>::iterator
#include "SmartPtrList.h"
#else
#define ArrayList(x) BinMessagePtrList
#define ArrayList_iterator(x) BinMessagePtrList::iterator
#include "BinMessagePtrList.h"
#endif

// Namespace
namespace org {
namespace himalay {
namespace ccif {

using namespace ::org::himalay::msgs::runtime;

// Constructor
diagnosticReq_1_Mmode::diagnosticReq_1_Mmode() {
    references= 0;


    // header
    org_himalay_ccif_APDUHeader__NEW( header);
    // length
    length= (ui8)0;
    // selfDiagnostics
    selfDiagnostics.setMemberSize(0);
}
// Destructor
diagnosticReq_1_Mmode::~diagnosticReq_1_Mmode() {
    // header
    org_himalay_ccif_APDUHeader__DELETE( header);
    // length

    // selfDiagnostics
    {
#ifndef org_himalay_ccif_diagnosticReq1mMosemsg_USE_SMART_PTR
        selfDiagnostics.deleteMembers();
#endif
    }
}

int diagnosticReq_1_Mmode::readNoHeader(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;

    // length
    {
        length=(istream.readUI8());
        retVal+=1;
    }
    // selfDiagnostics
    for (; istream.available() > 0 ; ) {
        org_himalay_ccif_diagnosticReq1mMosemsg__PTR_TYPE temp;
        org_himalay_ccif_diagnosticReq1mMosemsg__NEW( temp);
        retVal += temp->read(istream);
        selfDiagnostics.add(temp);
    }
    return retVal;
}

int diagnosticReq_1_Mmode::read(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;
    // read header
    retVal += header->read(istream);
    // read length
    {
        length=(istream.readUI8());
        retVal+=1;
    }
    // read selfDiagnostics
    for (; istream.available() > 0 ; ) {
        org_himalay_ccif_diagnosticReq1mMosemsg__PTR_TYPE temp;
        org_himalay_ccif_diagnosticReq1mMosemsg__NEW( temp);
        retVal += temp->read(istream);
        selfDiagnostics.add(temp);
    }

    return retVal;
}

int diagnosticReq_1_Mmode::write(DataOutputStream& ostream) { // throws IOException
    {   /** fix dependent sizes for header **/
    }

    {   /** fix dependent sizes for selfDiagnostics**/
    }

    int retVal= 0;
    // write header
    if ( header != NULL ) {
        retVal +=header->write(ostream);
    }
    // write length
    ostream.writeUI8(length);
    retVal +=1;
    // write selfDiagnostics
{   for (ArrayList_iterator(diagnosticReq1mMosemsg) it=  selfDiagnostics.getIterator() ; it.hasNext();) {
            {
                retVal +=it.get()->write(ostream);
            }
        }
    }

    return retVal;
}

int diagnosticReq_1_Mmode::dump(DumpContext& dc) { // throws IOException
    dc.indent();
    dc.getPs().print("diagnosticReq_1_Mmode\n");
    dc.increaseIndent();
    int retVal= 0;
    // write header
    if ( header != NULL ) {
        dc.indent();
        dc.getPs().println("header") ;
        retVal +=header->dump(dc);
    }
    // write length
    dc.indent();
    dc.getPs().print("length=");
    dc.getPs().println((long)length);
    // write selfDiagnostics
{   for (ArrayList_iterator(diagnosticReq1mMosemsg) it=  selfDiagnostics.getIterator() ; it.hasNext();) {
            {
                dc.indent();
                dc.getPs().println("element");
                retVal +=it.get()->dump(dc);
            }
        }
    }
    dc.decreaseIndent();
    return retVal;
}


// Getter for header
//org_himalay_ccif_APDUHeader__PTR_TYPE diagnosticReq_1_Mmode::getHeader()
//{
//return this->header;
//}

// Setter for header
//void diagnosticReq_1_Mmode::setHeader(org_himalay_ccif_APDUHeader__PTR_TYPE val)
//{
//this->header= val;
//}

// Getter for length
//ui8 diagnosticReq_1_Mmode::getLength()
//{
//return this->length;
//}

// Setter for length
//void diagnosticReq_1_Mmode::setLength(ui8 val)
//{
//this->length= val;
//}

// Getter for selfDiagnostics
//ArrayList(diagnosticReq1mMosemsg) diagnosticReq_1_Mmode::getSelfDiagnostics()
//{
//return this->selfDiagnostics;
//}

// Setter for selfDiagnostics
//void diagnosticReq_1_Mmode::setSelfDiagnostics(ArrayList(diagnosticReq1mMosemsg) val)
//{
//this->selfDiagnostics= val;
//}

int diagnosticReq_1_Mmode::getSize() { // throws IOException
    int retVal = 0;

    return retVal;
}

void diagnosticReq_1_Mmode::setHeader(org_himalay_ccif_APDUHeader__PTR_TYPE header)
{
    this->header= header;
}

org_himalay_ccif_APDUHeader__PTR_TYPE diagnosticReq_1_Mmode::getHeader()
{
    return this->header;
}

#ifdef org_himalay_ccif_diagnosticReq_1_Mmode__USE_SMART_PTR
void intrusive_ptr_add_ref(org::himalay::ccif::diagnosticReq_1_Mmode* p) {
    p->increaseRef();
};
void intrusive_ptr_release(org::himalay::ccif::diagnosticReq_1_Mmode* p) {
    p->decreaseRef();
};;
#endif


}
}
}

// End of code